lib/active_record/connection_adapters/spanner_adapter.rb

```diff
@@ -198,7 +198,7 @@ def initialize_type_map m = type_map
 
       def register_array_types m
         m.register_type %r{^ARRAY<BOOL>}i, Type::Spanner::Array.new(Type::Boolean.new)
-        m.register_type %r{^ARRAY<BYTES\((MAX|d+)\)>}i, Type::Spanner::Array.new(Type::Binary.new)
+        m.register_type %r{^ARRAY<BYTES\((MAX|d+)\)>}i, Type::Spanner::Array.new(ActiveRecord::Type::Spanner::Bytes.new)
```

Comment:
Why is this change needed? Won't this possibly break existing usage of `ARRAY<BYTES>`?

Reply:
This change was needed because `ActiveRecord::Type::Binary#serialize` returns `ActiveRecord::Type::Binary::Data` .

https://github.com/rails/rails/blob/83217025a171593547d1268651b446d3533e2019/activemodel/lib/active_model/type/binary.rb#L24

Without this change, these tests will fail:

```
1) Error:
ActiveRecord::Type::AllTypesTest#test_update_record:
ArgumentError: A value of type ActiveModel::Type::Binary::Data is not supported.
  /Users/yasutomo.uemori/.asdf/installs/ruby/3.1.1/lib/ruby/gems/3.1.0/gems/google-cloud-spanner-2.13.0/lib/google/cloud/spanner/convert.rb:111:in `object_to_grpc_value'
  /Users/yasutomo.uemori/.asdf/installs/ruby/3.1.1/lib/ruby/gems/3.1.0/gems/google-cloud-spanner-2.13.0/lib/google/cloud/spanner/convert.rb:97:in `block in object_to_grpc_value'
  /Users/yasutomo.uemori/.asdf/installs/ruby/3.1.1/lib/ruby/gems/3.1.0/gems/google-cloud-spanner-2.13.0/lib/google/cloud/spanner/convert.rb:97:in `map'
  /Users/yasutomo.uemori/.asdf/installs/ruby/3.1.1/lib/ruby/gems/3.1.0/gems/google-cloud-spanner-2.13.0/lib/google/cloud/spanner/convert.rb:97:in `object_to_grpc_value'
  /Users/yasutomo.uemori/ghq/src/github.com/googleapis/ruby-spanner-activerecord/lib/activerecord_spanner_adapter/base.rb:127:in `block in _create_grpc_values_for_insert'
  /Users/yasutomo.uemori/ghq/src/github.com/googleapis/ruby-spanner-activerecord/lib/activerecord_spanner_adapter/base.rb:126:in `map'
  /Users/yasutomo.uemori/ghq/src/github.com/googleapis/ruby-spanner-activerecord/lib/activerecord_spanner_adapter/base.rb:126:in `_create_grpc_values_for_insert'
  /Users/yasutomo.uemori/ghq/src/github.com/googleapis/ruby-spanner-activerecord/lib/activerecord_spanner_adapter/base.rb:60:in `_insert_record'

  2) Error:
ActiveRecord::Type::AllTypesTest#test_create_record:
ArgumentError: A value of type ActiveModel::Type::Binary::Data is not supported.
    /Users/yasutomo.uemori/.asdf/installs/ruby/3.1.1/lib/ruby/gems/3.1.0/gems/google-cloud-spanner-2.13.0/lib/google/cloud/spanner/convert.rb:111:in `object_to_grpc_value'
    /Users/yasutomo.uemori/.asdf/installs/ruby/3.1.1/lib/ruby/gems/3.1.0/gems/google-cloud-spanner-2.13.0/lib/google/cloud/spanner/convert.rb:97:in `block in object_to_grpc_value'
    /Users/yasutomo.uemori/.asdf/installs/ruby/3.1.1/lib/ruby/gems/3.1.0/gems/google-cloud-spanner-2.13.0/lib/google/cloud/spanner/convert.rb:97:in `map'
    /Users/yasutomo.uemori/.asdf/installs/ruby/3.1.1/lib/ruby/gems/3.1.0/gems/google-cloud-spanner-2.13.0/lib/google/cloud/spanner/convert.rb:97:in `object_to_grpc_value'
    /Users/yasutomo.uemori/ghq/src/github.com/googleapis/ruby-spanner-activerecord/lib/activerecord_spanner_adapter/base.rb:127:in `block in _create_grpc_values_for_insert'
    /Users/yasutomo.uemori/ghq/src/github.com/googleapis/ruby-spanner-activerecord/lib/activerecord_spanner_adapter/base.rb:126:in `map'
    /Users/yasutomo.uemori/ghq/src/github.com/googleapis/ruby-spanner-activerecord/lib/activerecord_spanner_adapter/base.rb:126:in `_create_grpc_values_for_insert'
    /Users/yasutomo.uemori/ghq/src/github.com/googleapis/ruby-spanner-activerecord/lib/activerecord_spanner_adapter/base.rb:60:in `_insert_record'
```

Reply:
Ah. Thanks for the explanation, that makes sense to me.